pkg/api/pod/util.go

```diff
@@ -1260,3 +1261,56 @@ func MarkPodProposedForResize(oldPod, newPod *api.Pod) {
 		}
 	}
 }
+
+// KEP: https://kep.k8s.io/4639
+func dropImageVolumes(podSpec, oldPodSpec *api.PodSpec) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) || imageVolumesInUse(oldPodSpec) {
+		return
+	}
+
+	imageVolumeNames := sets.New[string]()
+	var newVolumes []api.Volume
+	for _, v := range podSpec.Volumes {
+		if v.Image != nil {
+			imageVolumeNames.Insert(v.Name)
+			continue
+		}
+		newVolumes = append(newVolumes, v)
+	}
+	podSpec.Volumes = newVolumes
+
+	dropVolumeMounts := func(givenMounts []api.VolumeMount) (newVolumeMounts []api.VolumeMount) {
+		for _, m := range givenMounts {
+			if !imageVolumeNames.Has(m.Name) {
+				newVolumeMounts = append(newVolumeMounts, m)
+			}
+		}
+		return newVolumeMounts
+	}
```

Comment:
No change needed here, but thinking to myself that in this case, I'm not certain this is kinder than letting the pod fail validation.

Reply:
I'm more for simplicity and decided to remove that code + tests from [c5a8de6 ](https://github.com/kubernetes/kubernetes/commit/c5a8de67a91d0de996ac39dbd625285d60e97537)to [eecdd1d](https://github.com/kubernetes/kubernetes/commit/eecdd1ddfe0f8aeee5b0693eb8ae17c20fad2847)

Reply:
> No change needed here, but thinking to myself that in this case, I'm not certain this is kinder than letting the pod fail validation.

Miscommunication here.  There are two parts

1. Dropping the volume itself.  This must be done.
2. Dropping the mounts that reference that volume.  It helps make the pod valid, it probably doesn't make the pod succeed overall but it passes validation.  I could go either way on this part, so I wasn't hoping for a change.  On detailed reflection, I can imagine writing a cross-compatible podspec with a binary that looks in location/A for data before downloading it from another source, so this should be restored.

Reply:
Makes sense, I restored the code.